### ISSUE TITLE
query text is cut off within postgres_exporter

### ISSUE THREAD

**Issue by mzwettler2** [CrunchyData/postgres-operator#3527]
the query text is cut off within postgres_exporter / grafana. 

this makes it hard or even impossible to inspect suspect queries.

question: any way to get the whole query text from there?

**Comment by mzwettler2** [comment]
I found that the queries are already cutted by the postgres_exporter queries.

Is there any reason why you did that?


ccp_pg_stat_statements_top_max:
  query: "SELECT pg_get_userbyid(s.userid) as role,
              d.datname AS dbname,
              s.queryid,
              btrim(replace(left(s.query, 40), '\n', '')) AS query,
              s.max_exec_time AS exec_time_ms
          FROM public.pg_stat_statements s
          JOIN pg_catalog.pg_database d
          ON d.oid = s.dbid
          ORDER BY 5 DESC
          LIMIT #PG_STAT_STATEMENTS_LIMIT#"

**Comment by mzwettler2** [comment]
@benjaminjb 
May I ask you for your input?
Do you have any hint for the following behavior?


what I did:
=======
1. I reworked "queries_pg_stat_statements.yml" and removed all these "left(s.query,40)" clauses:
=> btrim(replace(s.query, '\n', '')) AS query
2. I rebuild the postgres_exporter container and copied-in the reworked yamls:
=> COPY ./queries_pg_stat_statements.yml /opt/cpm/conf/pg13/queries_pg_stat_statements.yml
3. I also enabled full debugging (CRUNCHY_DEBUG=true, postgres_exporter --log.level=debug)


what I got:
=======
1. postgres_exporter was started
2. the reworked queries were copied to "/tmp/queries.yaml"
3. **postgres_exporter did not generate any of the "ccp_" metrics anymore** (only the standard "pg_" metrics were still generated)
4. postgres_exporter simply ignored all "ccp_" metrics
5. postgres_exporter did not generate an error or any log or debug message telling why it ignored the "ccp_" metrics


Any idea?

**Comment by tony-landreth** [comment]
Hi @mzwettler2! The queryid column on the Query Statistics dashboard can normally be used to retrieve the full query from pg_stat_statements. Unfortunately, there may be a bug in that dashboard, causing it to sometimes present inaccurate queryid's. We'll be looking into that.

**Comment by mzwettler2** [comment]
> Hi @mzwettler2! The queryid column on the Query Statistics dashboard can normally be used to retrieve the full query from pg_stat_statements. Unfortunately, there may be a bug in that dashboard, causing it to sometimes present inaccurate queryid's. We'll be looking into that. Thanks for asking the question!

@tony-landreth 
Yes, but only if it is still included in pg_stat_statements. The later I look, the less likely it is to find the statement there. That's the problem.

**Comment by tony-landreth** [comment]
@mzwettler2 Good point! We do have something in our backlog to address this. Thanks for the reminder!

**Comment by tfidfwastaken** [comment]
Hi @tony-landreth!

We seem to have bumped into this same issue on our setup—do you have any recommendations for a workaround? Or perhaps if you have thoughts on how to address this upstream, we could try to see if we can be of any help.

**Comment by mzwettler2** [comment]
any news?

**Comment by j0nn1K** [comment]
Have update on this topic?
